In [1]:
import pandas as pd
from scipy import stats

Q1) F&B manager wants to determine whether there is any significant difference in
the diameter of the cutlet between two units. A randomly selected sample of cutlets
was collected from both units and measured? Analyze the data and draw inferences at
5% significance level. Please state the assumptions and tests that you carried out to
check validity of the assumptions.
 Minitab File : Cutlets.mtw


In [2]:
df=pd.read_csv('Cutlets.csv')
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [3]:
Unit_A=pd.Series(df['Unit A'])
Unit_B=pd.Series(df['Unit B'])

In [4]:
stats.ttest_ind(Unit_A,Unit_B)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

Q2) A hospital wants to determine whether there is any difference in the average
Turn Around Time (TAT) of reports of the laboratories on their preferred list.
They collected a random sample and recorded TAT for reports of 4 laboratories.
TAT is defined as sample collected to report dispatch.
   
Analyze the data and determine whether there is any difference in average TAT 
among the different laboratories at 5% significance level.


In [5]:
df=pd.read_csv('labTAT.csv')
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [6]:
Laboratory1=pd.Series(df['Laboratory 1'])
Laboratory2=pd.Series(df['Laboratory 2'])
Laboratory3=pd.Series(df['Laboratory 3'])
Laboratory4=pd.Series(df['Laboratory 4'])

In [7]:
stats.f_oneway(Laboratory1,Laboratory2,Laboratory3,Laboratory1)

F_onewayResult(statistic=64.94851005731616, pvalue=3.2318872286662227e-35)

Q3) Sales of products in four different regions is tabulated for males and females.
Find if male-female buyer rations are similar across regions.

In [8]:
df=pd.read_csv('BuyerRatio.csv')
df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [9]:
df_table=df.iloc[:,1:6]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [10]:
df_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [11]:
val=stats.chi2_contingency(df_table)
val

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

Here, p_value = 0.6603094907091882Since, p_Value(0.66030949) > aplha(0.05), We will accept null hypothesis.
Inference: Male-female buyer rations are similar across regions.

Q4) TeleCall uses 4 centers around the globe to process customer order forms.
They audit a certain %  of the customer order forms. Any error in order form
renders it defective and has to be reworked before processing.  The manager
wants to check whether the defective %  varies by centre. Please analyze the data
at 5% significance level and help the manager draw appropriate inferences


In [12]:
df=pd.read_csv('Costomer+OrderForm.csv')
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [13]:
df.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [14]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [15]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [16]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [18]:
import numpy as np

In [19]:
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [21]:
# Chi2 contengency independence test
stats.chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Here, p_value = 0.2771020991233135 Since p_value(0.2771020991233135) > alpha(0.05), Accept null hypothesis.
Inference: Proportion of defective % across the center is same.